modules/javafx.graphics/src/main/java/com/sun/glass/ui/ios/IosRobot.java

```diff
@@ -25,76 +25,91 @@
 
 package com.sun.glass.ui.ios;
 
-import com.sun.glass.ui.*;
-import java.nio.IntBuffer;
+import javafx.scene.input.KeyCode;
+import javafx.scene.input.MouseButton;
+import javafx.scene.paint.Color;
+
+import com.sun.glass.ui.Application;
+import com.sun.glass.ui.GlassRobot;
 
 /**
  * iOS platform implementation class of test automation Robot.
  */
-final class IosRobot extends Robot {
+final class IosRobot extends GlassRobot {
 
     private long ptr = 0;
 
     // init and create native robot object
     private native long _init();
-    @Override protected void _create() {
+
+    @Override public void create() {
+        Application.checkEventThread();
```

Comment:
What this method will check? If it throws an exception then probably it should be specified and followed by all implementations.

Reply:
This is specified in the public API for Robot:

```
+     * @throws IllegalStateException if this object is constructed on a thread
+     * other than the JavaFX Application Thread.
```

Reply:
I understood that this method is not a "public API" in terms of JavaFX specification. But if this method would have complete specification then it will be obvious that all implementations should throw an exception "if this object is constructed on a thread other than the JavaFX Application Thread." But some of implementations are empty and cannot throw any exception. I am not sure is it a bug or not.

Reply:
Not a bug since the platform-independent code also checks. This matches what Glass platforms do for other calls (not just Robot). I suppose it could be done more cleanly / consistently, but I don't want to hold up getting this in for this cleanup.

Reply:
ok